code/modules/mob/mob.dm

```diff
@@ -639,9 +639,30 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
 		return 1
 
 	face_atom(A)
-	var/list/result = A.examine(src)
+	var/list/result
+	if(client)
+		LAZYINITLIST(client.recent_examines)
+		var/ref_to_atom = ref(A)
+		var/examine_time = client.recent_examines[ref_to_atom]
+		if(examine_time && (world.time - examine_time < EXAMINE_MORE_WINDOW))
+			result = A.examine_more(src)
+			if(!length(result))
+				result += "<span class='notice'><i>You examine [A] closer, but find nothing of interest...</i></span>"
+		else
+			result = A.examine(src)
+			client.recent_examines[ref_to_atom] = world.time // set to when we last normal examine'd them
+			addtimer(CALLBACK(src, PROC_REF(clear_from_recent_examines), ref_to_atom), RECENT_EXAMINE_MAX_WINDOW)
```

Comment:
Rather than timers to automatically clear the entry, I think it's fine to keep it as is and just prune old entries on the next examine by looping through the list (or not prune at all but eh)

Reply:
agree with mochi here, timers are just gonna be more overhead

just make a proc or a loop that goes through recent_examines and call `clear_from_recent_examines(ref_to_clear)` on any refs with an old enough world.time val